### ISSUE TITLE
DeprecationWarning of pyarrow occurs when accessing to HDFS

### ISSUE THREAD

**Issue by nai62** [pfnet/pfio#139]
When I access to HDFS using pfio, the following DeprecationWarning occurs.

```
/..../pfio/pfio/filesystems/hdfs.py:153: DeprecationWarning: pyarrow.hdfs.connect is deprecated as of 2.0.0, please use pyarrow.fs.HadoopFileSystem instead.
```

## Versions
pfio==1.0.1
pyarrow==2.0.0

**Comment by kuenishi** [comment]
With pyarrow 3.0.0, it doesn't reproduce. Is it still issue for you?
```
$ pip install --user pfio
Collecting pfio
  Using cached pfio-1.0.1-py3-none-any.whl (30 kB)
Collecting pyarrow
  Downloading pyarrow-3.0.0-cp36-cp36m-manylinux2014_x86_64.whl (20.7 MB)
     |████████████████████████████████| 20.7 MB 11.9 MB/s
Requirement already satisfied: numpy>=1.16.6 in ./.local/lib/python3.6/site-packages (from pyarrow->pfio) (1.17.0)
Installing collected packages: pyarrow, pfio
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts
.
hsr-bullet 0.1.0 requires pyyaml==5.1, but you have pyyaml 3.13 which is incompatible.
Successfully installed pfio-1.0.1 pyarrow-3.0.0
WARNING: You are using pip version 20.3.1; however, version 21.0.1 is available.
You should consider upgrading via the '/usr/bin/python3 -m pip install --upgrade pip' command.
⚓=mn1g2 🐍=3.6.8  kota@jm00z0cm07:~$
$
⚓=mn1g2 🐍=3.6.8  kota@jm00z0cm07:~$
$
⚓=mn1g2 🐍=3.6.8  kota@jm00z0cm07:~$
$ python3
Python 3.6.8 (default, Jan 14 2019, 11:02:34)
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pfio
>>> pfio.open('hdfs:///user/kota/hello.txt')
21/02/08 07:38:49 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
<_io.BufferedReader>
>>> pfio.open('hdfs:///user/kota/hello.txt').read()
b'hello hadoop\n'
>>> from pyarrow import hdfs
>>> hdfs.connect()
<pyarrow.hdfs.HadoopFileSystem object at 0x7fe9e04c5bc8>
>>> hdfs.connect().open('hello.txt')
hdfsOpenFile(hello.txt): FileSystem#open((Lorg/apache/hadoop/fs/Path;I)Lorg/apache/hadoop/fs/FSDataInputStream;) error:
RemoteException: File does not exist: /user/hdfs/hello.txt
```

Also on 2.0.0 , it didn't reproduce for me.
```
$ pip3 install --user pyarrow==2.0.0
Collecting pyarrow==2.0.0
  Downloading pyarrow-2.0.0-cp36-cp36m-manylinux2014_x86_64.whl (17.7 MB)
     |████████████████████████████████| 17.7 MB 14.5 MB/s
Requirement already satisfied: numpy>=1.14 in ./.local/lib/python3.6/site-packages (from pyarrow==2.0.0) (1.17.0)
Installing collected packages: pyarrow
  Attempting uninstall: pyarrow
    Found existing installation: pyarrow 3.0.0
    Uninstalling pyarrow-3.0.0:
      Successfully uninstalled pyarrow-3.0.0
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
hsr-bullet 0.1.0 requires pyyaml==5.1, but you have pyyaml 3.13 which is incompatible.
Successfully installed pyarrow-2.0.0
WARNING: You are using pip version 20.3.1; however, version 21.0.1 is available.
You should consider upgrading via the '/usr/bin/python3 -m pip install --upgrade pip' command.
⚓=mn1g2 🐍=3.6.8  kota@jm00z0cm07:~$
$ python3
Python 3.6.8 (default, Jan 14 2019, 11:02:34)
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from pyarrow import hdfs
>>> import pfio
>>> pfio.open('hdfs:///user/kota/hello.txt').read()
21/02/08 07:41:43 WARN util.NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
b'hello hadoop\n'
```

Do you have any clue on it?

**Comment by nai62** [comment]
It doesn't reproduce for me as well even with the same versions and the environment I used. Though I'm not really sure what was the cause, let me close this issue.
Thank you for your help!

**Comment by nai62** [comment]
Hi, I've noticed this issue has not been resolved. The DeprecationWarning seems to be somehow hidden in the stderr, but it still occurs and is captured by pytest, for example.

## Reproduction code
```
import warnings
warnings.simplefilter('always')

import pfio
pfio.open('hdfs:///user/hirano/hello.txt').read()
```

Output (stderr):
```
/work/.pyenv/versions/anaconda3-2020.07/lib/python3.8/site-packages/pfio/filesystems/hdfs.py:152: DeprecationWarning: pyarrow.hdfs.connect is deprecated as of 2.0.0, please use pyarrow.fs.HadoopFileSystem instead.
  connection = hdfs.connect()
```

## Versions
Python 3.8.3
pfio==1.4.0
pyarrow==3.0.0

**Comment by kuenishi** [comment]
I did a quick survey to update to [PyArrow's new API](http://arrow.apache.org/docs/python/generated/pyarrow.fs.HadoopFileSystem.html). Unfortunately it doesn't support several important functionalities for us, e.g. directory listing. c.f. https://github.com/apache/arrow/blob/release-4.0.0/cpp/src/arrow/filesystem/hdfs.h

I have no other idea but to keep using legacy API.

**Comment by kuenishi** [comment]
This issue will be addressed by v2 API in next release, 1.5.0. It will be slightly different from current API around `HdfsFileStat` properties.
